code/game/objects/effects/spawners/structure.dm

```diff
@@ -22,288 +22,258 @@ again.
-
-/obj/effect/spawner/structure/window/hollow/plasma/corner
-	icon_state = "phwindow_spawner_corner_se"
-	spawn_list = 	list(/obj/structure/grille, /obj/structure/window/plasma, /obj/structure/window/plasma/spawner/east)
-
-/obj/effect/spawner/structure/window/hollow/plasma/corner/northeast
-	icon_state = "phwindow_spawner_corner_ne"
-	spawn_list = 	list(/obj/structure/grille, /obj/structure/window/plasma/spawner/north, /obj/structure/window/plasma/spawner/east)
-
-/obj/effect/spawner/structure/window/hollow/plasma/corner/northwest
-	icon_state = "phwindow_spawner_corner_nw"
-	spawn_list = 	list(/obj/structure/grille, /obj/structure/window/plasma/spawner/north, /obj/structure/window/plasma/spawner/west)
-
-/obj/effect/spawner/structure/window/hollow/plasma/corner/southwest
-	icon_state = "phwindow_spawner_corner_sw"
-	spawn_list = 	list(/obj/structure/grille, /obj/structure/window/plasma, /obj/structure/window/plasma/spawner/west)
+	spawn_list = list(/obj/structure/grille, /obj/structure/window/plasma, /obj/structure/window/plasma/spawner/north, /obj/structure/window/plasma/spawner/east, /obj/structure/window/plasma/spawner/west)
 
 /obj/effect/spawner/structure/window/hollow/plasma/end
-	icon_state = "phwindow_spawner_end_s"
-	spawn_list = 	list(/obj/structure/grille, /obj/structure/window/plasma, /obj/structure/window/plasma/spawner/east, /obj/structure/window/plasma/spawner/west)
-
-/obj/effect/spawner/structure/window/hollow/plasma/end/north
-	icon_state = "phwindow_spawner_end_n"
-	spawn_list = 	list(/obj/structure/grille, /obj/structure/window/plasma/spawner/north, /obj/structure/window/plasma/spawner/east, /obj/structure/window/plasma/spawner/west)
-
-/obj/effect/spawner/structure/window/hollow/plasma/end/east
-	icon_state = "phwindow_spawner_end_e"
-	spawn_list = 	list(/obj/structure/grille, /obj/structure/window/plasma, /obj/structure/window/plasma/spawner/north, /obj/structure/window/plasma/spawner/east)
-
-/obj/effect/spawner/structure/window/hollow/plasma/end/west
-	icon_state = "phwindow_spawner_end_w"
-	spawn_list = 	list(/obj/structure/grille, /obj/structure/window/plasma, /obj/structure/window/plasma/spawner/north, /obj/structure/window/plasma/spawner/west)
+	icon_state = "phwindow_spawner_end"
+
+/obj/effect/spawner/structure/window/hollow/plasma/end/Initialize()
+	switch(dir)
+		if(NORTH)
+			spawn_list = list(/obj/structure/grille, /obj/structure/window/plasma/spawner/north, /obj/structure/window/plasma/spawner/east, /obj/structure/window/plasma/spawner/west)
+		if(EAST)
+			spawn_list = list(/obj/structure/grille, /obj/structure/window/plasma, /obj/structure/window/plasma/spawner/north, /obj/structure/window/plasma/spawner/east)
+		if(SOUTH)
+			spawn_list = list(/obj/structure/grille, /obj/structure/window/plasma, /obj/structure/window/plasma/spawner/east, /obj/structure/window/plasma/spawner/west)
```

Comment:
South east and west all in two lines?

Reply:
Correct. It's an end piece.

They look like 
![image](https://user-images.githubusercontent.com/6209658/30516113-78ebe77c-9b03-11e7-9c0f-46b7ab20a4db.png)